README.md
#Accordiom - The Ultra Tiny Accordion

Just 1.7kB minified, and yes it's spelt with an 'm'. Single file jQuery plugin and no external stylesheet required.

[View the demo](http://www.simonboak.co.uk/accordiom/)
 

##Get Started

###1. Include jQuery and accordiom.js
```html
<script type="text/javascript" src="http://code.jquery.com/jquery-1.9.1.min.js"></script>
<script type="text/javascript" src="js/accordiom.js"></script>
```

###2. HTML markup
```html
<div id="accordion1">
	<div class="accordionButton"><h2>Item 1</h2></div>
	<div class="accordionContent">
		<p>Item 1 Content</p>
	</div>
	
	<div class="accordionButton"><h2>Item 2</h2></div>
	<div class="accordionContent">
		<p>Item 2 Content</p>
	</div>
	
	<div class="accordionButton"><h2>Item 3</h2></div>
	<div class="accordionContent">
		<p>Item 3 Content</p>
	</div>
</div>
```

** The h2 tag isn't even needed - any HTML can be in the accordionButton and accordionContent and the divs can be replaced with any other block level HTML5 element that would make more sense in your site.

###3. JavaScript
```html
<script type="text/javascript">
	$(document).ready( function () {
		$('#accordion1').accordiom();
	});
</script>
```


##Full Options


```javascript
$('#accordion1').accordiom({
	speed: 500, // Set the animation duration. 0 gives no animation
	showFirstItem: false, // Either show or hide the first item when loading
	beforeChange: function (accordionButton) {}, // Callback function called on click of the accordionButton before an item is hidden or shown
	afterChange: function (accordionButton) {}, // Callback function called after change of item
	onLoad: function (accordionButton) {} // Callback function called once the accordion is set up and event listeners in place
});
```

##Callbacks
2 callback functions are defined in the options above. The argument passed to the function is the div with class .accordionButton that was clicked on and can be used like this:
```javascript
$('#accordion1').accordiom({
	afterChange: function (accordionButton) {
		$(accordionButton).next('.accordionContent').append('Opened!');
	}
});
```
The example above appends the text 'Opened!' to the content of the content div for the opened item.

The class .on is also added to the active item so it can be styled and targeted in scripts.


##Functions
From version 0.3, Accordiom includes 3 functions that can be useful when you wish to change the state of the accordion through JavaScript. They are used as follows:

```javascript
$().accordiom.openAll('#accordion1'); // Open all accordion items
$().accordiom.closeAll('#accordion1'); // Close the content of all items
$().accordiom.openItem('#accordion1', 2); // Open item of specified number (zero indexed) so example will open 3rd item
```


[![githalytics.com alpha](https://cruel-carlota.pagodabox.com/123151625a476883f1b3e163df4cb001 "githalytics.com")](http://githalytics.com/simonboak/Accordiom)

changelog.txt
Version 0.3 14 Mar 2013
- Added openAll, closeAll and openItem helper functions

Version 0.2 8 Feb 2013
- Added onLoad callback

Version 0.1.1 7 Feb 2013
- Fixed bug where if first item was set to be open but then clicked to close it, it would close and then reopen.

Version 0.1 First release


js/old versions/accordiom-0.1.1.js
/*!
 * accordiom.js version 0.1.1
 * http://github.com/simonboak/accordiom
 * Public Domain
 *
 */

(function($){
    $.accordiom = function (el, options) {
        // To avoid scope issues, use 'base' instead of 'this'
        // to reference this class from internal events and functions.
        var base = this;
        
        // Access to jQuery and DOM versions of element
        base.$el = $(el);
        base.el = el;
        
        // Add a reverse reference to the DOM object
        base.$el.data('accordiom', base);
        
        base.init = function () {
            base.options = $.extend({}, $.accordiom.defaultOptions, options);
            
            // Put your initialization code here
        };
        
        // Sample Function, Uncomment to use
        // base.functionName = function(paramaters){
        // 
        // };
        
        // Run initializer
        base.init();
    };
    
    $.accordiom.defaultOptions = {
        speed: 500,
        showFirstItem: true,
        beforeChange: function () {},
        afterchange: function () {}
    };
    
    $.fn.accordiom = function (options) {
        return this.each( function () {
            (new $.accordiom(this, options));
            
            // For some reason the default options weren't being extended so I'll do it manually
            if (!options) {
	            options = $.accordiom.defaultOptions;
            }
            
            // Hide the content, but conditionally leave the first one open
            if (options.showFirstItem) {
                $(this).children('.accordionContent').not(':first').hide();
                $(this).children('.accordionButton').first().addClass('on');
            } else {
                $(this).children('.accordionContent').hide();
            }
            
            // Bind events to the buttons
            $(this).children('.accordionButton').on('click', function () {
            
                // Grab the container element (which would be used in the initial selector)
                var $selectorEl = $(this).parent('div');

                if (options.beforeChange) {
                    options.beforeChange.call(this, this); // sends the clicked accordion button element
                }

                if ($(this).is('.on')) {
                    $selectorEl.children('.accordionContent').slideUp(options.speed);
                    $(this).removeClass('on');
                } else {
                    $selectorEl.children('.accordionContent').slideUp(options.speed);
                    $selectorEl.children('.accordionButton').removeClass('on');
                    $(this).next('.accordionContent').slideDown(options.speed);
                    $(this).addClass('on');
                }
                
                if (options.afterChange) {
                    options.afterChange.call(this, this); // sends the clicked accordion button element
                }
            });
            
        });
    };
    
})(jQuery);

js/accordiom.js
/*!
 * accordiom.js version 0.2
 * http://github.com/simonboak/accordiom
 * Public Domain
 *
 */

(function($){
    $.accordiom = function (el, options) {
        // To avoid scope issues, use 'base' instead of 'this'
        // to reference this class from internal events and functions.
        var base = this;
        
        // Access to jQuery and DOM versions of element
        base.$el = $(el);
        base.el = el;
        
        // Add a reverse reference to the DOM object
        base.$el.data('accordiom', base);
        
        base.init = function () {
            base.options = $.extend({}, $.accordiom.defaultOptions, options);
            
            // Put your initialization code here
        };
        
        // Sample Function, Uncomment to use
        // base.functionName = function(paramaters){
        // 
        // };
        
        // Run initializer
        base.init();
    };
    
    $.accordiom.defaultOptions = {
        speed: 500,
        showFirstItem: true,
        beforeChange: function () {},
        afterchange: function () {},
        onLoad: function () {}
    };
    
    $.fn.accordiom = function (options) {
        return this.each( function () {
            (new $.accordiom(this, options));
            
            // For some reason the default options weren't being extended so I'll do it manually
            if (!options) {
	            options = $.accordiom.defaultOptions;
            }
            
            // Hide the content, but conditionally leave the first one open
            if (options.showFirstItem) {
                $(this).children('.accordionContent').not(':first').hide();
                $(this).children('.accordionButton').first().addClass('on');
            } else {
                $(this).children('.accordionContent').hide();
            }
            
            // Fire the onLoad callback once all's set up
            if (options.onLoad) {
	            options.onLoad.call(this, this);
            }
            
            // Bind events to the buttons
            $(this).children('.accordionButton').on('click', function () {
            
                // Grab the container element (which would be used in the initial selector)
                var $selectorEl = $(this).parent('div');

                if (options.beforeChange) {
                    options.beforeChange.call(this, this); // sends the clicked accordion button element
                }

                if ($(this).is('.on')) {
                    $selectorEl.children('.accordionContent').slideUp(options.speed);
                    $(this).removeClass('on');
                } else {
                    $selectorEl.children('.accordionContent').slideUp(options.speed);
                    $selectorEl.children('.accordionButton').removeClass('on');
                    $(this).next('.accordionContent').slideDown(options.speed);
                    $(this).addClass('on');
                }
                
                if (options.afterChange) {
                    options.afterChange.call(this, this); // sends the clicked accordion button element
                }
            });
            
        });
    };
    
})(jQuery);
